xchange-core/src/main/java/org/knowm/xchange/service/marketdata/MarketDataService.java

```diff
@@ -177,4 +178,26 @@ default Trades getTrades(Instrument instrument, Object... args) throws IOExcepti
   default Trades getTrades(Params params) throws IOException {
     throw new NotYetImplementedForExchangeException("getTrades");
   }
+
+
+  /**
+   * Get the CandleStickData for given currency between startDate to endDate.
+   *
+   * @param currencyPair currencyPair.
+   * @param startDate startDate of candle stick, E.G. march 2022 (Epoch timestamp in milis).
+   * @param endDate endDate of candle stick, E.G. april 2022 (Epoch timestamp in milis).
+   * @param period period of candle stick in milis.
+   * @param args extra exchange specific parameters.
+   * @return The CandleStickData, null if some sort of error occurred. Implementers should log the error.
+   * @throws ExchangeException - Indication that the exchange reported some kind of error with the
+   * request or response
+   * @throws NotAvailableFromExchangeException - Indication that the exchange does not support the
+   * requested function or data
+   * @throws NotYetImplementedForExchangeException - Indication that the exchange supports the
+   * requested function or data, but it has not yet been implemented
+   * @throws IOException - Indication that a networking error occurred while fetching JSON data
+   */
+  default CandleStickData getCandleStickData(CurrencyPair currencyPair, String startDate, String endDate, String period, Object... args) throws IOException {
```

Comment:
These should not be string, startDate and endDate should be Date and period should be int.